src/Microsoft.AspNetCore.Rewrite/Internal/IISUrlRewrite/UrlRewriteFileParser.cs

```diff
@@ -165,16 +185,24 @@ private void ParseConditions(XElement conditions, UrlRewriteRuleBuilder builder,
         private void ParseCondition(XElement condition, UrlRewriteRuleBuilder builder, PatternSyntax patternSyntax)
         {
             bool ignoreCase;
-            if (!bool.TryParse(condition.Attribute(RewriteTags.IgnoreCase)?.Value, out ignoreCase))
+            if (condition.Attribute(RewriteTags.IgnoreCase) == null)
```

Comment:
for each attribute, you retrieve it three times (though on different code paths). store condition.Attribute(...) in a local variable.